File: solutions/beecrowd/1221/1221.cpp
#include <cmath>
#include <cstdint>
#include <iostream>

bool is_prime(std::uint32_t n) {
    if (n == 2) {
        return true;
    }
    if (n % 2 == 0 || n < 2) {
        return false;
    }

    for (std::uint32_t i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    std::int32_t n, p;

    std::cin >> n;

    while (n--) {
        std::cin >> p;

        if (is_prime(p)) {
            std::cout << "Prime" << std::endl;
        } else {
            std::cout << "Not Prime" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/1221/1221.py
import math
import sys

input = sys.stdin.readline
output = sys.stdout.write


def is_prime(n):
    if n == 2:
        return True

    if n % 2 == 0 or n < 2:
        return False

    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False

    return True


n = int(input())

for _ in range(n):
    p = int(input())

    if is_prime(p):
        output('Prime\n')
    else:
        output('Not Prime\n')
